Maco/catalogue/views.py

```diff
@@ -38,6 +45,20 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class ProductSearch(ListView):
+    template_name = 'catalogue/product_list_page.jinja'
+    model = Product
+
+    def get_queryset(self):
+        queryset = super().get_queryset().has_stock_records()
+        return ProductFilter(self.request.GET, queryset).qs.order_by('stockrecords__price_excl_tax')
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['filter'] = ProductFilter(self.request.GET, self.get_queryset())
```

Comment:
This is calling product filter w/ the restults of product filter - I think break the filter into a method on the class - something like

```suggestion
        def get_filter(self):
           queryset = super().get_queryset().has_stock_records()
           return ProductFilter(self.request.GET, queryset)
         def get_queryset(self):
           filter = self.get_filter()
           return filter.qs.order_by(..)
         ...
        context['filter'] = self.get_filter()
```